dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonUtils1_1.java

```diff
@@ -0,0 +1,294 @@
+
+        for (Pair p : pairs) {
+
+            if (p instanceof KeyValuePair) {
+
+                kvp = (KeyValuePair) p;
+                key = kvp.getKey();
+                value = kvp.getValue();
+
+                if (key.equals(LogFieldConstants.IBM_QUERYSTRING)) {
+
+                    String jsonQueryString = value;
+                    if (jsonQueryString != null) {
+                        try {
+                            jsonQueryString = URLDecoder.decode(jsonQueryString, "UTF-8");
+                        } catch (UnsupportedEncodingException e) {
+                            // ignore, use the original value;
+                        }
+                    }
+                    isFirstField = isFirstField & !CollectorJsonHelpers.addToJSON(sb, key, jsonQueryString, false, true, false, isFirstField, kvp);
+
+                } else if (key.equals(LogFieldConstants.IBM_USERAGENT)) {
+
+                    String userAgent = value;
+
+                    if (userAgent != null && userAgent.length() > MAX_USER_AGENT_LENGTH) {
+                        userAgent = userAgent.substring(0, MAX_USER_AGENT_LENGTH);
+                    }
+
+                    isFirstField = isFirstField & !CollectorJsonHelpers.addToJSON(sb, key, userAgent, false, false, false, isFirstField, kvp);
+
+                } else if (key.equals(LogFieldConstants.IBM_DATETIME)) {
+
+                    String datetime = CollectorJsonHelpers.dateFormatTL.get().format(Long.parseLong(value));
+                    isFirstField = isFirstField & !CollectorJsonHelpers.addToJSON(sb, key, datetime, false, true, false, isFirstField, kvp);
+
+                } else {
+
+                    isFirstField = isFirstField & !CollectorJsonHelpers.addToJSON(sb, key, value, false, true, false, isFirstField, kvp);
+
+                }
+            }
+        }
+
+        if (tags != null) {
+            addTagNameForVersion(sb).append(CollectorJsonHelpers.jsonifyTags(tags));
+        }
+
+        sb.append("}");
+
+        return sb.toString();
+    }
+
+    private static String jsonifyTraceAndMessage(int maxFieldLength, String wlpUserDir,
+                                                 String serverName, String hostName, String eventType, Object event, String[] tags) {
+
+        GenericData genData = (GenericData) event;
+        StringBuilder sb = new StringBuilder();
+        boolean isFirstField = true;
+        ArrayList<Pair> pairs = genData.getPairs();
+        KeyValuePair kvp = null;
+        String key = null;
+        String value = null;
+
+        sb.append("{");
+
+        isFirstField = CollectorJsonHelpers.addCommonFields(sb, hostName, wlpUserDir, serverName, isFirstField, eventType);
+
+        for (Pair p : pairs) {
+
+            if (p instanceof KeyValuePair) {
+
+                kvp = (KeyValuePair) p;
+                key = kvp.getKey();
+                value = kvp.getValue();
+
+                if (!(key.equals(LogFieldConstants.SEVERITY))) {
+
```

Comment:
would be easier to understand method overall if you did this...
if (key.equals(LogFieldConstants.SEVERITY))) {}
else if (key.equals(...